src/dbnode/storage/shard.go

```diff
@@ -842,14 +844,21 @@ func (s *dbShard) writeAndIndex(
 			return ts.Series{}, false, err
 		}
 	} else {
-		// This is an asynchronous insert and write
+		// This is an asynchronous insert and write which means we need to clone the annotation
+		// because it could get returned to a pool at any point after this function returns.
+		annotationClone := s.opts.BytesPool().Get(len(annotation))
```

Comment:
you don't need to always do this. can you put it behind a nil check. i.e. 

```
var annotationClone checked.Bytes
if len(annotation) != 0 {
// ...
}
```

Reply:
maybe make it a helper function (this method is long enough as is)

Reply:
thanks I had a mental todo for this but forgot